subsys/fs/nvs/nvs.c

```diff
@@ -361,26 +355,28 @@ static void _nvs_sector_advance(struct nvs_fs *fs, u32_t *addr)
 	}
 }
 
-/* allocation entry close (this closes the current sector) by writing all
- * zeros at fs->ate_wra.
+/* allocation entry close (this closes the current sector) by writing offset
+ * of last ate to the sector end.
  */
 static int _nvs_sector_close(struct nvs_fs *fs)
 {
 	int rc;
-	u8_t buf[sizeof(struct nvs_ate)];
+	struct nvs_ate close_ate;
 	size_t ate_size;
 
 	ate_size = _nvs_al_size(fs, sizeof(struct nvs_ate));
 
-	(void)memset(buf, 0, ate_size);
-
-	rc = _nvs_flash_al_wrt(fs, fs->ate_wra, buf, sizeof(struct nvs_ate));
-	if (rc) {
-		return rc;
-	}
+	close_ate.id = 0xFFFF;
```

Comment:
Is this id documented somewhere?

Reply:
No, but it is also not important. The ate at the end of the sector is used to contain the jump to the last ate added. This ate at the end is treated differently than other ate's. It could also be used to store a version of the nvs file system. It does not limit the id's used by nvs, they can still be in the range 0x0000-0xFFFF.

Reply:
thanks for explanation